templates/Dockerfile.erb

```diff
@@ -17,22 +17,30 @@ ENV GEM_HOME /fluentd/vendor/bundle/ruby/2.6.0
 ENV FLUENTD_DISABLE_BUNDLER_INJECTION 1
 
 COPY Gemfile* /fluentd/
-  RUN buildDeps="sudo make gcc g++ libc-dev libffi-dev<% if target == "graylog" %> build-essential patch zlib1g-dev liblzma-dev<% elsif target == "kafka" %> build-essential autoconf automake libtool pkg-config<% end %><% if requires_git %> git<% end %>" \
-     && apt-get update \
-     && apt-get upgrade -y \
-     && apt-get install \
-     -y --no-install-recommends \
-     $buildDeps net-tools \
-    && gem install bundler --version 1.16.2 \
-    && bundle config silence_root_warning true \
-    && bundle install --gemfile=/fluentd/Gemfile --path=/fluentd/vendor/bundle \
-    && SUDO_FORCE_REMOVE=yes \
-    apt-get purge -y --auto-remove \
-                  -o APT::AutoRemove::RecommendsImportant=false \
-                  $buildDeps \
- && rm -rf /var/lib/apt/lists/* \
-    && gem sources --clear-all \
-    && rm -rf /tmp/* /var/tmp/* /usr/lib/ruby/gems/*/cache/*.gem
+
+RUN buildDeps="sudo make gcc g++ libc-dev libffi-dev<% if target == "graylog" %> build-essential patch zlib1g-dev liblzma-dev \
+<% elsif target == "kafka" %> build-essential autoconf automake libtool pkg-config python <% end %><% if requires_git %> git<% end %>" \
+runtimeDeps="<% if target == "kafka" %>krb5-kdc libsasl2-modules-gssapi-mit libsasl2-dev <% end %> " \
```

Comment:
Please move to `<% elsif target == "kafka" %>`

Reply:
I understand it seems redundant but I did it this way in case there are additional buildDeps for other targets in the future or if other targets have some runtimeDeps in the future. Thought it would be easier to modify this way. 

We are also installing the libraries under $buildDeps $runtimeDeps net-tools (lines 27-29). Otherwise, will need an if block there to include the runtimeDeps only for kafka (even though it'll still work if the $runDeps variable does not exist technically).

Thoughts? Should I still continue to move it to <% elsif target == "kafka" %>?

Reply:
Hi @repeatedly ,

We're trying to contribute the mechanics for adding runtime dependencies like libsasl that needs to be present with librdkafka for getting the Kerberos Authentification working with Kafka.

As the builddeps are removed as expected at the end, any way to keep the librairies present after?

Regards

Reply:
:)

Reply:
Thanks, feedback is appreciated as always :)